hack/setup_env.sh

```diff
@@ -15,9 +19,20 @@ function print_help() {
   echo "Usage: bash ${0} (${ALL_FUNCS})"
 }
 
+function download() {
+    url=$1
+    output=$2
+    cached_filename=$(echo -n "$url" | sha256sum | awk '{ print $1 }')
+    cache_full_path="${CACHE_DIR}/${cached_filename}"
+    [ -f "${cache_full_path}" ] || curl --retry 5 -L "$url" -o "$cache_full_path"
```

Comment:
By doing this we will leverage buildkit cache, and avoid downloading the file again if we already have a file with the same name in the cache directory

Reply:
Can you add what you said as a comment in the source file?